### ISSUE TITLE
Unexpected '(' in "service : (opt InternetIdentityInit) -> {

### ISSUE THREAD

**Issue by nmattia** [nomeata/haskell-candid#16]
Hi there,

I'm trying to update the candid definition used in Internet Identity. Unfortunately, candid doesn't seem to appreciate some argument:

```
backend-tests.hs:74:37: error:
    • Candid service:101:11:
    |
101 | service : (opt InternetIdentityInit) -> {
    |           ^
unexpected '('
expecting "/*", "//", white space, or {

    • In the quasi-quotation:
        [Candid.candid|
...
```

Is this expected? can we expect support for this in a future candid version?

**Comment by nomeata** [comment]
Yes, exactly. One could argue that these init arguments shouldn't even be there, but that's not a helpful stance. I should at least make the parser ignore them.

Traveling right now though, so don't expect a quick fix. Hope that's ok. PRs accepted.

**Comment by nmattia** [comment]
No rush! Thanks for the confirmation and safe travels. :)
